lib/apiservers/engine/backends/archive.go

```diff
@@ -250,12 +250,9 @@ func NewArchiveStreamWriterMap(mounts []types.MountPoint, containerDestPath stri
 		// Here, we must tell the portlayer to remove "mnt/A".  The key to determining whether to
 		// strip "A" or "mnt/A" is based on the container destination path.
 		if containerDestPath != "/" && strings.HasPrefix(aw.mountPoint.Destination, containerDestPath) {
-			aw.filterSpec.StripPath = strings.TrimPrefix(aw.mountPoint.Destination, containerDestPath)
+			aw.filterSpec = vicarchive.GenerateFilterSpec(containerDestPath, aw.mountPoint, !strings.Contains(aw.mountPoint, containerDestPath), vicarchive.CopyTo)
```

Comment:
woops yeah actually. it needs to be more than that for sure since aw.mountpoint is not correct. Sorry about that for some reason my IDE did not catch that. Additionally, it must only be called against mounts that are involved with the operation. if you look any one of those targets that you supplied would be a primary target, but it you are doing a `write` then ---> /mnt/volA and /mnt/VolB are not involved in the operation at all and thus a writer/reader should never be initialized for them. if you are doing a read you get the same thing. I might have made a false assumption about where the filter spec should occur.